tests/neg/i21212.scala

```diff
@@ -0,0 +1,11 @@
+//> using options -source 3.7
+object Minimization:
+
+  trait A
+  trait B extends A
+
+  def test1(using a1: A)(using    b1: B) = summon[A] // picks (most general) a1
+  def test2(using a2: A)(implicit b2: B) = summon[A] // error: ambiguous
```

Comment:
Should this not be un-ambiguous under 3.7 (`a2` is the most general) ?

(It is in the version incorporating the latest fixes to `healAmbiguous`)

Reply:
No because one is an implicit and the other is a given. So each compares against the other with their own rules.

Reply:
I am going to try to do a quick fix for the problems we encountered.

Reply:
> I am going to try to do a quick fix for the problems we encountered.

@odersky I have something here for `healAmbiguous`

Reply:
But the changes in that PR make this test (tests/neg/121212.scala) no longer ambiguous

Reply:
Oh my bad, they in fact do **not**, I had simply forgotten to use `-source:3.7`
I will add the test back